programs/compressed-pda/src/instructions.rs

```diff
@@ -124,60 +155,56 @@ pub struct TransferInstruction<'info> {
     pub invoking_program: Option<UncheckedAccount<'info>>,
 }
 
+/// collects invocations without proofs
+/// invocations are collected and processed when an invocation with a proof is received
 #[account]
 pub struct CpiSignatureAccount {
-    pub signatures: Vec<CpiSignature>,
+    pub slot: u64,
+    pub signatures: Vec<InstructionDataTransfer>,
 }
 
-#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize)]
-pub struct CpiSignature {
-    pub program: Pubkey,
-    pub tlv_hash: [u8; 32],
-    pub tlv_data: TlvDataElement,
-}
-
-// TODO: parse utxos a more efficient way, since owner is sent multiple times this way
 #[derive(Debug)]
 #[account]
```

Comment:
the struct InstructionDataTransfer is now used by the `cpiSignatureAccount`, so tagging it as anchor account breaks the IDL.
instead, we can tag it as regular type with AnchorDeserialize,also allows us to not have to prepend 8 bytes when deserializing and can use ::deserialize instead of try_deserialize_unchecked()